### PR TITLE
zabbix5: dbtls support

### PR DESCRIPTION
**Pull request by aclarkee**
Adds new zabbix server & proxy parameters for TLS connection to the database

Does not deploy the cafile - that is left as a manual task for now.

### CODE REVIEW

**Review by igalic — CHANGES_REQUESTED** [review]
I'd like to see some spec tests for this

**Comment by aclarkee** [comment]
> I'd like to see some spec tests for this

Could you give some guidance on spec test design, how to test them etc? I have zero experience in writing spec tests, unfortunately. That seems to be where most of the current pull requests are sitting at - I would like to contribute, but do not know how to contribute these test specs.

**Comment by igalic** [comment]
a good way to start writing spec tests is by copying an existing test in the general vicinity of the parameters you're modifying, and go from there

we're also available to help you in #voxpupuli on freenode, or on the puppet slack

File: manifests/proxy.pp [hunk 1]
```diff
@@ -108,6 +108,17 @@
 # [*database_port*]
```
Comment (igalic) [inline manifests/proxy.pp:108]: the permissions on this file change
that's probably a mistake

File: manifests/server.pp [hunk 2]
```diff
@@ -83,6 +83,17 @@
 # [*database_port*]
```
Comment (igalic) [inline manifests/server.pp:83]: the permissions on this file change
that's probably a mistake

File: manifests/web.pp [hunk 3]
```diff
@@ -215,6 +215,8 @@
   $database_password                                                  = $zabbix::params::server_database_password,
```
Comment (igalic) [inline manifests/web.pp:215]: the permissions on this file change
that's probably a mistake

File: templates/zabbix_proxy.conf.erb [hunk 4]
```diff
@@ -124,6 +124,8 @@ DBUser=<%= @database_user %>
 #	Comment this line if no password is used.
```
Comment (igalic) [inline templates/zabbix_proxy.conf.erb:124]: the permissions on this file change
that's probably a mistake

File: templates/zabbix_server.conf.erb [hunk 5]
```diff
@@ -105,6 +105,8 @@ DBUser=<%= @database_user %>
 #	Comment this line if no password is used.
 #
```
Comment (igalic) [inline templates/zabbix_server.conf.erb:106]: the permissions on this file change
that's probably a mistake

**Review by igalic — CHANGES_REQUESTED** [review]
👀

File: manifests/proxy.pp (hunk 1 above)
Comment (aclarkee) [reply manifests/proxy.pp:108]: Good pickup - thanks. I'll send through a commit to fix this later today.

**Pull request closed without merging**